src/XIVLauncher.Core/Components/SettingsPage/Tabs/SettingsTabDalamud.cs

```diff
@@ -29,26 +29,28 @@ public SettingsTabDalamud()
                      return;
                  }
 
-                 if (Directory.Exists(Program.Config.DalamudManualInjectPath) && Directory.GetFiles(Program.Config.DalamudManualInjectPath).FirstOrDefault(x => x == "Dalamud.Injector.exe") is not null)
+                 if (Program.Config.DalamudManualInjectPath is not null &&
+                    Program.Config.DalamudManualInjectPath.Exists &&
+                    Program.Config.DalamudManualInjectPath.GetFiles().FirstOrDefault(x => x.Name == Program.DALAMUD_INJECTOR_NAME) is not null)
                  {
-                     Program.DalamudUpdater.RunnerOverride = new FileInfo(Path.Combine(Program.Config.DalamudManualInjectPath, Program.DALAMUD_INJECTOR_NAME));
-                 }
-             }),
+                     Program.DalamudUpdater.RunnerOverride = new FileInfo(Path.Combine(Program.Config.DalamudManualInjectPath.FullName, Program.DALAMUD_INJECTOR_NAME));
+                }
+            }),
 
-            new SettingsEntry<string>("Manual Injection Path", "The path to the local version of Dalamud where Dalamud.Injector.exe is located", () => Program.Config.DalamudManualInjectPath, (input) =>
+            new SettingsEntry<DirectoryInfo>("Manual Injection Path", "The path to the local version of Dalamud where Dalamud.Injector.exe is located", () => Program.Config.DalamudManualInjectPath, (input) =>
             {
                 Program.Config.DalamudManualInjectPath = input;
-                Program.DalamudUpdater.RunnerOverride = new FileInfo(Path.Combine(input, Program.DALAMUD_INJECTOR_NAME));
+                Program.DalamudUpdater.RunnerOverride = new FileInfo(Path.Combine(input!.FullName, Program.DALAMUD_INJECTOR_NAME));
```

Comment:
This will crash if the value of the input is empty, even if the entry is hidden. Here's a quick fix:
```suggestion
                if (input is not null)
                    Program.DalamudUpdater.RunnerOverride = new FileInfo(Path.Combine(input!.FullName, Program.DALAMUD_INJECTOR_NAME));
```